Add spacing tokens mdx story

narin opened the pull request:
<!-- PR title naming convention:
'[Issue type] Brief summary of issue suitable for changelog or copy/paste issue title',
where Issue type = bug, feature, spike, CU (code upkeep), etc.-->

<!-- Preferred branch naming convention:
'[Issue type]/[Issue #]-[Your name]-[Summary of issue]',
where Issue type = bug, feature, spike, CU (code upkeep), etc.-->

## Description of Change
<!-- Describe the change and context with which it was made beyond ACs unless straightforward.
Consider:
 - What is relevant to code reviewer(s) and not in the ticket?
 - What context may be relevant to a future dev or you in 6 months about this PR?
 - Did the course of work lead to notable dead ends? If so, why didn't they pan out?
 - Did the change add new dependencies? Why?
 - Were there important sources to link? Examples: an open bug with a dependency project, an article of someone else solving the same problem that was partially or wholly copied, external documentation relevant to solution
 -->
Adds a Spacing tokens storybook page with a table of tokens and visual example, similar to our color tokens page and [VADS's docs](https://design.va.gov/foundation/spacing-units#primitive).

#### Testing Packages
<!-- List or range of alpha/beta packages published in association with this PR, if any -->
N/A

### Screenshots/Video
<!-- Add screenshots or video as needed; before/after recommended if appropriate. 
Convenience side-by-side formatting:
Before/after: <img src="" width="49%" />&nbsp;&nbsp;<img src="" width="49%" />
Accordion before/after: <details><summary>Before/after</summary><img src="" width="49%" />&nbsp;&nbsp;<img src="" width="49%" /></details>
-->
<img width="1343" alt="image" src="https://github.com/user-attachments/assets/364d5c0a-deaa-4244-a70c-8927d2c971de">




## Testing
<!-- Describe testing conducted to validate changes.
Consider highlighting:
- What testing was not explicitly done and may be relevant for QA? 
- Edge cases validated
- Special situations that could not be tested
- Any testing performed in a consuming app -->
- [x] Tested on Web

## PR Checklist
Code reviewer validation:
- General
	- [x] PR is linked to ticket(s)
	- [x] PR has `changelog` label applied if it's to be included in the changelog
	- [x] Acceptance criteria: 
		- All satisfied _or_
		- Documented reason for not being performed _or_
		- Split to separate ticket and ticket is linked by relevant AC(s)
	- [x] Above PR sections adequately filled out
	- [x] If any breaking changes, in accordance with the [pre-1.0.0 versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library#versioning-policy): a CU ticket has been created for the VA Mobile App detailing necessary adjustments with the package version that will be published by this ticket
- Code
	- [x] Tests are included if appropriate (or split to separate ticket)
	- [x] New functions have proper TSDoc annotations

## Publish
<!-- Most changes entail a version increment; section can be removed for PRs exclusively within non-ship-relevant files (e.g. unit tests, Storybook stories) -->
If changes warrant a new version [per the versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library/blob/main/documentation/versioning.md) and the PR is approved and ready to merge:
- [ ] Merge `main` into branch
- [ ] Merge branch to `main`
- [ ] Verify that [Check Component Integrations](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/check-component-integrations.yml) workflow ran successfully
- [ ] [Publish new version](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/publish.yml) after merging

TimRoe commented on packages/components/storybook/spacing.stories.mdx:
```diff
@@ -0,0 +1,13 @@
+import { Meta } from '@storybook/blocks'
+
+import { SpacingTable } from './SpacingTable'
+
+<Meta title="Design Tokens/Spacing" />
+
```
Might be a special case of getting rid of the whitespace lines here? Looking at the image, it looks like there's actually whitespace above the Spacing heading which we probably don't want

TimRoe replied:
Or maybe we leave it, since the Colors token page also has it.

narin replied:
Test this out and removing the whitespace has no effect on the whitespace above

TimRoe reviewed (approved):
Looks good to me. 

One potential enhancement might be somewhat matching to the Colors page by centering the content and keeping doc alignment at the top in the same width as the token content.

jessicawoodin reviewed (changes requested):
Overall, this looks good, but I have some nitpicky feedback:
- It looks like the copy in the screenshot is different than what was provided in the outline. Can you add this copy and the Primitive header above the tokens?
> Spacing tokens are used to maintain a consistent rhythm when applied to margins, padding, and other dimensions.
> 
> [View guidance for spacing tokens on the VA Mobile Documentation Site](https://department-of-veterans-affairs.github.io/va-mobile-app/design/Foundation/Design%20tokens/Spacing)
> 
> **Primitive**

- Can you left align the Token/Value/Examples headers in the table?
- I liked Tim's idea to center the content like the Colors page. Right now, the text gets pretty wide on this Spacing page.

narin commented:
- Updated text
- Left aligned table column headings
- Changed table to 100% width
- 
@TimRoe @jessicawoodin Please take a look at the updated screenshot and let me know if there's anything else you'd like to see

jessicawoodin reviewed (approved):
Looks good to me.
